File: network-equipment-pms/pms-governance/src/main/java/com/dp/plat/governance/issue/entity/Issue.java
package com.dp.plat.governance.issue.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.dp.plat.common.entity.BaseEntity;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * Issue entity (governance three-books: issue log).
 *
 * <p>Tracks issues raised on a project. Issues can originate from materialized
 * risks or change requests, and can themselves be escalated to change requests.</p>
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("pms_issue")
public class Issue extends BaseEntity {

    /** Issue number (ISSUE-YYYY-XXXX). */
    @NotBlank(message = "问题编号不能为空")
    @Size(max = 50, message = "问题编号长度不能超过 50 个字符")
    private String issueNo;

    /** Project id. */
    @NotNull(message = "项目ID不能为空")
    private Long projectId;

    /** Issue description. */
    @NotBlank(message = "问题描述不能为空")
    @Size(max = 2000, message = "问题描述长度不能超过 2000 个字符")
    private String description;

    /** User id who raised the issue. */
    private Long raisedBy;

    /** User name who raised the issue. */
    @Size(max = 50, message = "提出人名称长度不能超过 50 个字符")
    private String raisedByName;

    /** Assignee user id. */
    private Long assigneeId;

    /** Assignee user name. */
    @Size(max = 50, message = "处理人名称长度不能超过 50 个字符")
    private String assigneeName;

    /** Priority (LOW, MEDIUM, HIGH, CRITICAL). */
    @NotBlank(message = "优先级不能为空")
    @Size(max = 20, message = "优先级长度不能超过 20 个字符")
    private String priority;

    /** Target resolution date. */
    private LocalDate targetResolveDate;

    /** Status (OPEN, IN_PROGRESS, RESOLVED, CLOSED). */
    @Size(max = 50, message = "状态长度不能超过 50 个字符")
    private String status;

    /** Source risk id (nullable, if from risk). */
    private Long sourceRiskId;

    /** Source risk number (nullable, if from risk). */
    @Size(max = 50, message = "来源风险编号长度不能超过 50 个字符")
    private String sourceRiskNo;

    /** Source change request id (nullable, if from change). */
    private Long sourceChangeId;

    /** Source change request number (nullable, if from change). */
    @Size(max = 50, message = "来源变更单号长度不能超过 50 个字符")
    private String sourceCrNo;

    /** Resolution time. */
    private LocalDateTime resolvedAt;

    /** Closure time. */
    private LocalDateTime closedAt;

    /** Resolution description (TEXT). */
    @Size(max = 2000, message = "解决方案长度不能超过 2000 个字符")
    private String resolution;
}

File: network-equipment-pms/pms-frontend/src/api/issue.ts
import { get, post } from '@/utils/request'
import type { ChangeRequestPriority } from './change-request'

/** 问题状态 */
export type IssueStatus = 'OPEN' | 'IN_PROGRESS' | 'RESOLVED' | 'CLOSED'

/** 问题实体 */
export interface Issue {
  id?: number
  issueNo?: string
  projectId: number
  description: string
  raisedBy?: number
  raisedByName?: string
  assigneeId?: number
  assigneeName?: string
  priority?: ChangeRequestPriority // LOW/MEDIUM/HIGH/CRITICAL
  targetResolveDate?: string
  status: IssueStatus
  sourceRiskId?: number
  sourceRiskNo?: string
  sourceChangeId?: number
  sourceChangeNo?: string
  createdAt?: string
  resolvedAt?: string
}

/** 问题列表查询参数 */
export interface IssueListQuery {
  page: number
  size: number
  projectId?: number
  status?: string
  assigneeId?: number
}

/** 分页结果 */
export interface IssueListResult {
  records: Issue[]
  total: number
}

export function listIssues(params: IssueListQuery): Promise<IssueListResult> {
  return get<IssueListResult>('/api/governance/issue', params)
}

export function createIssue(data: Issue): Promise<Issue> {
  return post<Issue>('/api/governance/issue', data)
}

export function assignIssue(id: number, assigneeId: number): Promise<Issue> {
  return post<Issue>(`/api/governance/issue/${id}/assign`, { assigneeId })
}

export function resolveIssue(id: number): Promise<Issue> {
  return post<Issue>(`/api/governance/issue/${id}/resolve`)
}

export function closeIssue(id: number): Promise<Issue> {
  return post<Issue>(`/api/governance/issue/${id}/close`)
}

export function escalateIssue(id: number, reason: string): Promise<boolean> {
  return post<boolean>(`/api/governance/issue/${id}/escalate`, { reason })
}
